File: backend/Services/Database/Session.cs
using System.ComponentModel.DataAnnotations;

namespace Services.Database
{
    public class Session
    {
        public int Id { get; set; }
        public int LanguageId { get; set; }
        public Language Language { get; set; } = null!;
        public int LevelId { get; set; }
        public Level Level { get; set; } = null!;
        public int Duration { get; set; } //minutes

        public string ChannelName { get; set; } = string.Empty;
        public int NumOfUsers { get; set; }

        public int CurrentNumOfUSers { get; set; } = 0;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<Tag> Tags { get; set; } = new List<Tag>();
    }
}

File: UI/justspeak_mobile/lib/models/session.dart
import 'package:frontend/models/tag.dart';
import 'package:json_annotation/json_annotation.dart';

part 'session.g.dart';

@JsonSerializable()
class Session {
  final int id;
  final String? language;
  final String? level;
  final int? languageId;
  final int? levelId;
  final int numOfUsers;
  final int currentNumOfUsers;
  final int duration;
  final DateTime createdAt;
  final String? channelName;
  final String? token;
  final List<Tag> tags;

  Session({
    required this.id,
    this.language = "",
    this.level = "",
    this.languageId,
    this.levelId,
    required this.numOfUsers,
    this.currentNumOfUsers = 0,
    required this.duration,
    required this.createdAt,
    this.channelName,
    this.token,
    required this.tags,
  });

  factory Session.fromJson(Map<String, dynamic> json) =>
      _$SessionFromJson(json);
  Map<String, dynamic> toJson() => _$SessionToJson(this);
}
